pkg/api/types.go

```diff
@@ -607,6 +607,10 @@ type SecretVolumeSource struct {
 	// the volume setup will error. Paths must be relative and may not contain
 	// the '..' path or start with '..'.
 	Items []KeyToPath `json:"items,omitempty"`
+	// Mode bits to use on created files by default. The used mode bits will
+	// be the provided AND 0777.
+	// Directories within the path are not affected by this setting
+	DefaultMode int32 `json:"defaultMode,omitempty"`
```

Comment:
https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#optional-vs-required

You're exactly right.  a default value of 0 for numbers or "" for strings is ambiguous when those values are valid for the field.